app/(playground)/p/[agentId]/contexts/execution.tsx

```diff
@@ -448,6 +448,10 @@ export function ExecutionProvider({
 			stepExecution: CompletedStepExecution,
 			artifact: TextArtifact,
 		) => void;
+		onFinishPerformExecution: (
+			endedAt: number,
+			durationMs: number,
+		) => Promise<void>;
```

Comment:
I think this should be Optional.

```suggestion
		onFinishPerformExecution?: (
			endedAt: number,
			durationMs: number,
		) => Promise<void>;
```